### PR TITLE
doc: reorder index

### PR DESCRIPTION
**Pull request by Alizter**
I've put the most important entries at the top and ordered the rest in alphabetical order.

### CODE REVIEW

**Comment by Alizter** [comment]
The comments I added create warnings so I will need to get rid of them.

File: doc/index.rst [hunk 1]
```diff
@@ -9,31 +9,94 @@ Welcome to Dune's Documentation!
 .. toctree::
    :maxdepth: 3
 
+   .. We include the titles of the pages here to make sure they are in
+      alphabetical order. Eventually we should name the files and titles
+      similarly.
+
+   .. This topics should appear first since they are the most important ones.
```
Comment (christinerose) [inline doc/index.rst:16]:
```suggestion
   .. These topics should appear first since they are the most important ones.
```

**Review by christinerose — APPROVED** [review]
Just one typo that I saw. This --> These

File: doc/index.rst (hunk 1 above)
Comment (Alizter) [reply doc/index.rst]: I had to get rid of the comments since they caused a warning.

**Comment by Alizter** [comment]
At some point we should rename the doc files themselves so that they match up with the titles. The titles also could use some work.

**Comment by christinerose** [comment]
> At some point we should rename the doc files themselves so that they match up with the titles. The titles also could use some work.

Agreed!

**Review by christinerose — APPROVED** [review]

**Comment by Alizter** [comment]
@rgrinberg Can you merge?

**Pull request merged**